### PR TITLE
docs: 修复章节chapter2.1 transformer架构

### PR DESCRIPTION
**Pull request by shiranstar0854**

### CODE REVIEW

**Review by linhongyu510 — APPROVED** [review]
这两处修改方向正确，可以消除“当前层同时接收上下层信号”的歧义，同时没有把所有前馈网络都误写成全连接网络：

- 第一处先保留 FNN“单向、无循环”的一般定义，再明确限定到“标准全连接 MLP”；
- 第二处与本章实际示例一致：`w1 = nn.Linear(dim, hidden_dim)`、`w2 = nn.Linear(hidden_dim, dim)`，数据按 `w1 -> ReLU -> w2 -> dropout` 前向传播；
- PyTorch 官方教程对 fully connected / linear layer 的契约也是“every input influences every output”，与改写后的相邻层连接描述一致。

我还做了三层竞争与历史核查：

- Issue #212 目前无评论、assignee 或其他关联实现；
- 开放 PR #192 虽然也修改同一章，但其 Chapter 2 补丁只涉及 Softmax、MLP 注释、LayerNorm、位置编码等其他位置，不触及本 PR 的两段；
- 历史 PR #165 / commit `723d618` 已把“全连接神经网络 = FNN”修正为“FNN 可通过全连接或特定方式传递”，但遗留了“和上下两层完全连接”的歧义，本 PR 是对该遗留问题的定向补全。

本地验证：

```text
git diff --check origin/main...d4cffb4            PASS
目标章节 main 中“上下两层”                    2 处
目标章节 PR head 中“上下两层”                 0 处
git merge-tree(base, origin/main, d4cffb4)    可生成
```

当前 head 没有 CI checks / commit statuses；上述是文档差异、实现契约和历史竞争核查，不把 Pages 部署或静态检查表述为运行时测试。非阻断建议在 PR 正文补充 `Fixes #212` 或 `Refs #212`，方便维护者和读者追踪问题来源。

本 Review 使用 AI 辅助梳理审计范围与验证命令；结论由我依据固定 commit `d4cffb4`、仓库源码、Git 历史及 PyTorch 官方文档逐项复核。
